source/common/network/utility.cc

```diff
@@ -5,6 +5,10 @@
 
 #if defined(__linux__)
 #include <linux/netfilter_ipv4.h>
+#ifndef IP6T_SO_ORIGINAL_DST
+// From linux/netfilter_ipv6/ip6_tables.h
+#define IP6T_SO_ORIGINAL_DST 80
```

Comment:
I think you're going to have to `#ifdef` for OS X in any case below, so might not be worth forcing this.